src/service/pluginService.ts

```diff
@@ -127,210 +112,147 @@ export class PluginService {
-      };
-      const data = new ExtensionMetadata(
-        meta.galleryApiUrl,
-        meta.id,
-        meta.downloadUrl,
-        meta.publisherId,
-        meta.publisherDisplayName,
-        meta.date
-      );
-      const info = new ExtensionInformation();
-      info.metadata = data;
-      info.name = ext.packageJSON.name;
-      info.publisher = ext.packageJSON.publisher;
-      info.version = ext.packageJSON.version;
-      list.push(info);
-    }
-
-    return list;
+    return vscode.extensions.all
+      .filter(ext => !ext.packageJSON.isBuiltin)
+      .map(ext => {
+        const meta = ext.packageJSON.__metadata || {
+          id: ext.packageJSON.uuid,
+          publisherId: ext.id,
+          publisherDisplayName: ext.packageJSON.publisher
+        };
+        const data = new ExtensionMetadata(
+          meta.galleryApiUrl,
+          meta.id,
+          meta.downloadUrl,
+          meta.publisherId,
+          meta.publisherDisplayName,
+          meta.date
+        );
+        const info = new ExtensionInformation();
+        info.metadata = data;
+        info.name = ext.packageJSON.name;
+        info.publisher = ext.packageJSON.publisher;
+        info.version = ext.packageJSON.version;
+        return info;
+      });
   }
 
   public static async DeleteExtension(
-    item: ExtensionInformation,
-    ExtensionFolder: string
+    extension: ExtensionInformation
   ): Promise<boolean> {
-    const destination = path.join(
-      ExtensionFolder,
-      item.publisher + "." + item.name + "-" + item.version
-    );
-
-    try {
-      await fs.remove(destination);
-      return true;
-    } catch (err) {
-      console.log("Sync : " + "Error in uninstalling Extension.");
-      console.log(err);
-      throw err;
-    }
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        vscode.commands
+          .executeCommand(
+            "workbench.extensions.uninstallExtension",
+            `${extension.publisher}.${extension.name}`
+          )
+          .then(resolve);
+      } catch (err) {
+        reject(err);
+        throw new Error(err);
+      }
+    });
   }
 
   public static async DeleteExtensions(
     extensionsJson: string,
-    extensionFolder: string,
     ignoredExtensions: string[]
   ): Promise<ExtensionInformation[]> {
-    const remoteList = ExtensionInformation.fromJSONList(extensionsJson);
-    const deletedList = PluginService.GetDeletedExtensions(
-      remoteList,
+    const remoteExtensions = ExtensionInformation.fromJSONList(extensionsJson);
+    const toDelete = PluginService.GetDeletedExtensions(
+      remoteExtensions,
       ignoredExtensions
     );
-    const deletedExt: ExtensionInformation[] = [];
+    const deletedExtensions: ExtensionInformation[] = [];
 
-    if (deletedList.length === 0) {
-      return deletedExt;
-    }
-    for (const selectedExtension of deletedList) {
+    toDelete.forEach(async selectedExtension => {

```

Comment:
we need to use `await` here.

Reply:
I'm not sure how to use await here, but we can use `Promise.all` to wait for the deletions to finish.